# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmCodegenTypes.cpp
# Commit: 1a73cfde72e8
# Full Hash: 1a73cfde72e8f4d19bdcfdbffa9fbc78bba01e24
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2021-09-28 09:54:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Introduce indirect stubs to optimize call_indirect. r=lth
#   
#   This patch introduces indirect stubs.
#   An indirect stub is a stub that takes care of any switching activities needed for call_indirect.
#   Before this patch, we have to conservatively assume that any call_indirect's target can be from a foreign instance,
# ==============================================================================

diff -r da3c5d8492db -r 1a73cfde72e8 js/src/wasm/WasmCodegenTypes.cpp
--- a/js/src/wasm/WasmCodegenTypes.cpp	Tue Sep 28 05:16:23 2021 +0000
+++ b/js/src/wasm/WasmCodegenTypes.cpp	Tue Sep 28 05:24:15 2021 +0000
@@ -139,7 +139,7 @@
   u.func.lineOrBytecode_ = 0;
   u.func.beginToUncheckedCallEntry_ = 0;
   u.func.beginToTierEntry_ = 0;
-  MOZ_ASSERT(isEntry());
+  MOZ_ASSERT(isEntry() || isIndirectStub());
   MOZ_ASSERT(begin_ <= end_);
 }
 